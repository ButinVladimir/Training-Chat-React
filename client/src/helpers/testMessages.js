import Message from './Message';

const testMessages = [
  new Message('1', '8/28/2018', 'User 1', [], 'Message 1'),
  new Message('2', '8/28/2018', 'User 1', [], 'Message 2'),
  new Message('3', '8/28/2018', 'User 1', ['User 2'], 'Message 3 to one user'),
  new Message('4', '8/28/2018', 'User 2', ['User 1', 'User 2'], 'User 2 answer'),
  new Message('5', '8/28/2018', 'User 3', ['User 1', 'User 2'], 'User 3 question?'),
  new Message('6', '8/28/2018', 'User 1', ['User 1', 'User 2', 'User 3'], 'User 1 answer'),
  new Message('7', '8/28/2018', 'User 2', [], 'Message 7'),
  new Message('8', '8/28/2018', 'User 2', [], 'Message 8'),
  new Message('9', '8/28/2018', 'User 1', [], 'Message 9'),
  new Message('10', '8/28/2018', 'User 3', [], 'Message 10'),
  new Message('11', '8/28/2018', 'User 1', [], 'Message 1'),
  new Message('12', '8/28/2018', 'User 1', [], 'Message 2'),
  new Message('13', '8/28/2018', 'User 1', ['User 2'], 'Message 3 to one user'),
  new Message('14', '8/28/2018', 'User 2', ['User 1', 'User 2'], 'User 2 answer'),
  new Message('15', '8/28/2018', 'User 3', ['User 1', 'User 2'], 'User 3 question?'),
  new Message('16', '8/28/2018', 'User 1', ['User 1', 'User 2', 'User 3'], 'User 1 answer'),
  new Message('17', '8/28/2018', 'User 2', [], 'Message 7'),
  new Message('18', '8/28/2018', 'User 2', [], 'Message 8'),
  new Message('19', '8/28/2018', 'User 1', [], 'Message 9'),
  new Message('20', '8/28/2018', 'User 3', [], 'Message 10'),
  new Message('21', '8/28/2018', 'User 1', [], 'Message 1'),
  new Message('22', '8/28/2018', 'User 1', [], 'Message 2'),
  new Message('23', '8/28/2018', 'User 1', ['User 2'], 'Message 3 to one user'),
  new Message('24', '8/28/2018', 'User 2', ['User 1', 'User 2'], 'User 2 answer'),
  new Message('25', '8/28/2018', 'User 3', ['User 1', 'User 2'], 'User 3 question?'),
  new Message('26', '8/28/2018', 'User 1', ['User 1', 'User 2', 'User 3'], 'User 1 answer'),
  new Message('27', '8/28/2018', 'User 2', [], 'Message 7'),
  new Message('28', '8/28/2018', 'User 2', [], 'Message 8'),
  new Message('29', '8/28/2018', 'User 1', [], 'Message 9'),
  new Message('30', '8/28/2018', 'User 3', [], 'Message 10'),
];

export default testMessages;
