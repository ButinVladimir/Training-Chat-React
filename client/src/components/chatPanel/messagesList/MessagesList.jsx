import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Socket } from 'socket.io-client';
import Message from './MessageContainer';
import { messageShape } from '../../../helpers/Message';
import './style.css';

class MessagesList extends Component {
  componentWillUnmount() {
    const { handlers, socket } = this.props;
    handlers.forEach((handler, event) => {
      socket.off(event, handler);
    });
  }

  render() {
    const { messages } = this.props;
    const convertedMessages = messages.map(message => (
      <Message key={message.id} {...message} />
    ));

    return (
      <div className="chat-messages-list-container">
        <ul className="chat-messages-list">
          {convertedMessages}
        </ul>
      </div>
    );
  }
}

MessagesList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(messageShape)).isRequired,
  handlers: PropTypes.instanceOf(Map).isRequired,
  socket: PropTypes.instanceOf(Socket).isRequired,
};

export default MessagesList;
