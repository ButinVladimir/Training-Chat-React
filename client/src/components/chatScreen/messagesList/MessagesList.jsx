import React from 'react';
import PropTypes from 'prop-types';
import Message from './Message';
import { messageShape } from '../../../helpers/Message';
import './style.css';

function MessagesList({ messages, onClickUser }) {
  const convertedMessages = messages.map(message => (
    <Message key={message.id} {...message} onClickUser={onClickUser} />
  ));

  return (
    <div className="chat-messages-list-container">
      <ul className="chat-messages-list">
        {convertedMessages}
      </ul>
    </div>
  );
}

MessagesList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(messageShape)).isRequired,
  onClickUser: PropTypes.func.isRequired,
};

export default MessagesList;
