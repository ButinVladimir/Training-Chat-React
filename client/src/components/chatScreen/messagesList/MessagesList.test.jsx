import React from 'react';
import { mount } from 'enzyme';
import MessagesList from './MessagesList';
import testMessages from '../../../helpers/testMessages';

describe('MessagesList', () => {
  it('renders without crashing', () => {
    const component = mount(<MessagesList messages={testMessages} onClickUser={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
