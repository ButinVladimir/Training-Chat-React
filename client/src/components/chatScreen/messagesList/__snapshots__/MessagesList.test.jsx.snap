// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList renders without crashing 1`] = `
<div
  className="chat-messages-list-container"
>
  <ul
    className="chat-messages-list"
  >
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 1
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 2
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        Message 3 to one user
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        User 2 answer
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 3: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        User 3 question?
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <button
        className="user-select to"
        key="User 3"
        type="button"
      >
        @User 3
      </button>
      <span
        className="message"
      >
        User 1 answer
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <span
        className="message"
      >
        Message 7
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <span
        className="message"
      >
        Message 8
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 9
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 3: 
      </button>
      <span
        className="message"
      >
        Message 10
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 1
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 2
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        Message 3 to one user
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        User 2 answer
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 3: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        User 3 question?
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <button
        className="user-select to"
        key="User 3"
        type="button"
      >
        @User 3
      </button>
      <span
        className="message"
      >
        User 1 answer
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <span
        className="message"
      >
        Message 7
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <span
        className="message"
      >
        Message 8
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 9
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 3: 
      </button>
      <span
        className="message"
      >
        Message 10
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 1
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 2
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        Message 3 to one user
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        User 2 answer
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 3: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <span
        className="message"
      >
        User 3 question?
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <button
        className="user-select to"
        key="User 1"
        type="button"
      >
        @User 1
      </button>
      <button
        className="user-select to"
        key="User 2"
        type="button"
      >
        @User 2
      </button>
      <button
        className="user-select to"
        key="User 3"
        type="button"
      >
        @User 3
      </button>
      <span
        className="message"
      >
        User 1 answer
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <span
        className="message"
      >
        Message 7
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 2: 
      </button>
      <span
        className="message"
      >
        Message 8
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 1: 
      </button>
      <span
        className="message"
      >
        Message 9
      </span>
    </li>
    <li
      className="chat-message"
    >
      <span
        className="date"
      >
        [8/28/2018]
      </span>
      <button
        className="user-select from"
        onClick={[Function]}
        type="button"
      >
        User 3: 
      </button>
      <span
        className="message"
      >
        Message 10
      </span>
    </li>
  </ul>
</div>
`;
